openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -4282,6 +4289,842 @@ def create_force(self, system, topology, **kwargs):
         system.addForce(gbsa_force)
 
 
+def validate_virtual_site_type_option(vsite_type):
+    """
+    Convert and validate the virtual site type specified in the VirtualSite element
+
+    Parameters
+    ----------
+    vsite_type : Any
+        The virtual site type to validate
+
+    Returns
+    -------
+    policy : str
+        The virtual site type if it is valid
+
+    Raises
+    ------
+    SMIRNOFFSpecError
+    ValueError
+    """
+
+    try:
+        vsite_type = str(vsite_type)
+    except ValueError:
+        raise
+
+    if vsite_type in [
+        "BondCharge",
+        "MonovalentLonePair",
+        "DivalentLonePair",
+        "TrivalentLonePair",
+    ]:
```

Comment:
It would be nice if this could be dynamically loaded from the find-able virtualsite subclasses (something like `[vst.TAGNAME for vst in all_subclasses(VirtualSiteHandler.VirtualSiteType)]`

Reply:
Done!